routing/astar_router.cpp

```diff
@@ -0,0 +1,218 @@
+#include "routing/astar_router.hpp"
+
+#include "routing/base/astar_algorithm.hpp"
+#include "routing/base/astar_progress.hpp"
+#include "routing/bicycle_directions.hpp"
+#include "routing/bicycle_model.hpp"
+#include "routing/car_model.hpp"
+#include "routing/features_road_graph.hpp"
+#include "routing/index_graph.hpp"
+#include "routing/pedestrian_model.hpp"
+#include "routing/route.hpp"
+#include "routing/turns_generator.hpp"
+
+#include "indexer/feature_altitude.hpp"
+#include "indexer/routing_section.hpp"
+
+#include "geometry/distance.hpp"
+#include "geometry/mercator.hpp"
+#include "geometry/point2d.hpp"
+
+#include "base/exception.hpp"
+
+using namespace routing;
+
+namespace
+{
+size_t constexpr kMaxRoadCandidates = 6;
+float constexpr kProgressInterval = 2;
+uint32_t constexpr kDrawPointsPeriod = 10;
+
+vector<Junction> ConvertToJunctions(IndexGraph const & graph, vector<Joint::Id> const & joints)
+{
+  vector<RoadPoint> const & fsegs = graph.RedressRoute(joints);
```

Comment:
There're a lot of |fseg(s)| names in our project. Probably it's not the best name for the value after the renaming.